src/components/dataset-viewer/Cell.vue

```diff
@@ -0,0 +1,75 @@
+<script lang="ts">
+import _ from 'lodash'
+import Vue from 'vue'
+
+import detectType from './detect-type'
+
+const defaultFormatter = (value: any, row: object, column: string): string =>
+  _.isObject(value) ? JSON.stringify(value) : _.toString(value)
+
+const CSS_CLASS = 'dataset-viewer__cell'
+
+type ClassModifierGenerator = (value: any, row: object, column: string, displayedValue: string) => string | undefined
+
+export default Vue.extend({
+  name: 'dataset-viewer-cell',
+
+  props: {
+    column: {},
+    row: {},
+    displayFormatted: {
+      default: false,
+      type: Boolean
+    },
+    formatter: {
+      default: defaultFormatter,
+      type: Function
+    },
+    classModifiers: {
+      default: () => [],
+      type: Array
+    },
+  },
+
+  functional: true,
+
+  render (createElement, ctx) {
+    let value = ctx.props.row[ctx.props.column]
+    let classes = [CSS_CLASS]
+    let type = detectType(value)
+
+    let displayedValue: string
+    if (ctx.props.displayFormatted) {
+      classes.push(`${CSS_CLASS}--formatted`)
+      displayedValue = ctx.props.formatter(value, ctx.props.row, ctx.props.column)
+    } else {
+      displayedValue = defaultFormatter(value, ctx.props.row, ctx.props.column)
+    }
+
+    classes = classes.concat(
```

Comment:
purely matter of style, but what do you think of `classes.push(...ctx.props.classModifiers.map(/* ... */))`? It might be _sligthly_ more consistent with the other `push` instructions above. (I let you choose to ignore this :))

Reply:
and then use `const` instead of `let` at variable definition :-)